docs/deployment.md

```diff
@@ -0,0 +1,237 @@
+run on each (unless static files are stored in a shared location).
+
+[Django's documentanion on static files]: https://docs.djangoproject.com/en/5.1/ref/contrib/staticfiles/#collectstatic
+[collectstatic management command]: https://docs.djangoproject.com/en/5.1/ref/contrib/staticfiles/#collectstatic
+[WhiteNoise]: https://whitenoise.readthedocs.io/en/stable/django.html
+
+### Creating Users
+
+At time of writing the app uses the built-in Django authentication system. This requires
+user accounts to be created before the user can login. The Django admin interface can be
+used to create users. The command `python manage.py createsuperuser` can be used to
+create a superuser account. This account can then be used to create other users via the
+admin interface.
+
+## Deploying with Docker Compose
+
+!!! note
+
+    The example deployment docker compose file is not actively tested and may not work
+    without modificaction. Nonetheless it provides a reasonable starting point for a
+    single server deployment.
+
+A simple and robust way to deploy the app on a single server is to use Docker Compose.
+This provides a way to define the services required by the app and flexibly manage
+configuration.
+
+An example Docker Compose file is provided in the `example_deployment` directory. You
+should be able to start the app with `docker compose -f
+example_deployment/docker-compose.yml up` and access the web app at `https://localhost`.
+
+!!! note
+
+    Although a kafka consumer service is defined in the example deployment file it will
+    not start successfully without `KAFKA_ADDRESS` being set to an operational Kafka
+    broker instance.
+
+The example deployment address the following previously mentioned considerations:
+
+- The software environment is provided is built from the `Dockerfile` in the root of the
+  repository.
+
+- The deloyment settings file is given by `example_deployment/deployment.py`. This is
+  mounted into the relevant containers at
+  `/usr/src/app/drunc_ui/settings/deployment.py`. The `DJANGO_SETTINGS_MODULE`
+  environment variable is then set to `drunc_ui.settings.deployment`.
+
+- The database is provided by a PostgreSQL container. The database name, user and
+  hostname are set in the deployment settings file. The database password is read from
+  the environment variable `DATABASE_PASSWORD` to facilitate secret management.
+
+- Environment variables are set in the `environment` section of each service in the
+  Docker Compose file.
+
+- Gunicorn is used as the WSGI server. The number of workers and threads is set in the
+  `command` section of the `app` service.
+
+- The web app is served over HTTPS by a reverse proxy. The `proxy` service uses [Caddy]
+  to forward traffic to the `app` service. Caddy is configured by
+  `example_deployment/Caddyfile` with is mounted into the container at
+  `/etc/caddy/Caddyfile`. Self signed certificates are used for HTTPS and mounted into
+  the `proxy` service.
```

Comment:
Might be worth stating the obvious here that the certificates in the example should not be used in production

Reply:
They actually couldn't be because they only work for localhost.